plugins/nsmserver/nsmserver_client.go

```diff
@@ -66,7 +66,14 @@ func (n nsmClientEndpoints) RequestDiscovery(ctx context.Context, cr *nsmconnect
 }
 
 func (n *nsmClientEndpoints) RequestAdvertiseChannel(ctx context.Context, cr *nsmconnect.ChannelAdvertiseRequest) (*nsmconnect.ChannelAdvertiseResponse, error) {
-	return nil, status.Error(codes.InvalidArgument, "Not Implemented...")
+	n.logger.Printf("received Channel advertisement... %+v", cr)
+	for _, c := range cr.NetmeshChannel {
+		n.logger.Printf("For NetworkService: %s channel: %s channel's socket location: %s", c.NetworkServiceName, c.Metadata.Name, c.SocketLocation)
```

Comment:
I don't think the 'For ' adds anything here, we should remove it and consider lower case network service